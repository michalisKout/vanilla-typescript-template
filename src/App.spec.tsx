import App from '@/App';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

describe('App', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<App />);
    expect(true).toBe(true);

    expect(baseElement).toMatchSnapshot();
  });

  it('should increment count', async () => {
    render(<App />);

    const button = document.querySelector('button');

    fireEvent.click(button!);

    await waitFor(() => {
      expect(screen.getByText('count is 1')).toBeInTheDocument();
    });
  });
});
