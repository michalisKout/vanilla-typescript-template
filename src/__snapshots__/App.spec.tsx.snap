// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should be true 1`] = `
<body>
  <div>
    <h1>
      React playground
    </h1>
    <div>
      <button>
        count is 
        0
      </button>
    </div>
  </div>
</body>
`;

exports[`App > should match snapshot 1`] = `
<body>
  <div>
    <h1>
      React playground
    </h1>
    <div>
      <button>
        count is 
        0
      </button>
    </div>
  </div>
</body>
`;
